File: Java_code/xor.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class XOR {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        boolean repeat = true;
        while (repeat) {
            System.out.print("Masukkan banyak angka: ");
            int n = in.nextInt();
            int a = 0;
            for (int i = 0; i < n; i++) {
                System.out.print("Masukkan angka: ");
                a ^= in.nextInt();
            }
            System.out.println("Hasil XOR dari "+n+" bilangan bulat tsb adalah: "+a);
            System.out.print("Ulangi lagi? (yes/no): ");
            String response = in.next();
            if (!response.equalsIgnoreCase("yes")) {
                repeat = false;
            }
        }
        System.out.println("END");
    }
}

File: GO_code/xor.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	var repeat bool = true
	for repeat {
		fmt.Print("Masukkan banyak angka: ")
		var n int
		fmt.Scan(&n)
		var a int = 0
		for i := 0; i < n; i++ {
			fmt.Print("Masukkan angka: ")
			var input string
			fmt.Scan(&input)
			num, err := strconv.Atoi(input)
			if err != nil {
				fmt.Println("Input tidak valid. Mohon masukkan bilangan bulat.")
				continue
			}
			a ^= num
		}
		fmt.Printf("Hasil XOR dari %d bilangan bulat tsb adalah: %d\n", n, a)
		var response string
		fmt.Print("Ulangi lagi? (yes/no): ")
		fmt.Scan(&response)
		if response != "yes" {
			repeat = false
		}
	}
	fmt.Println("END")
}